atlp-rwanda/stackup2-barefoot-backend
# #171380739 setup sequelize

#### What does this PR do?
Setup sequelize as a an ORM to the project
#### Description of Task to be completed?
- create a .sequelizerc file to hold configurations that sequelize-cli needs
- Install `sequelize` and its dependencies
- Install `sequelize-cli` for working with sequelize using command line
- Initialize sequelize with command `npx sequelize-cli init`
- Create a database with command `npx sequelize-cli db:create`
- Generate a dummy user model and migrations with command `npx sequelize-cli model:generate --name User --attributes email:string,password:string`
- Run user migration with command `npx sequelize-cli db:migrate`
#### How should this be manually tested?
- After cloning the project repo, run `npm i` inside the cloned repo folder
- Make sure to rename .env.example to .env, 
- In .env file, provide all values for variables under `#Development database`  and `#General`
- Finally, run the following commands respectively:
`git fetch`
`git checkout ch-setup-sequelize-171380733`
`npx sequelize-cli db:create` to create a database
`npx sequelize-cli db:migrate` to create a user table
- If the above steps have been followed correctly, you should see a new user table created in your database
#### Any background context you want to provide?
Refer to [the sequelize-cli](https://github.com/sequelize/cli/tree/master/docs) and [sequelize](https://sequelize.org/v5/) docs for more.
#### What are the relevant pivotal tracker stories?
#171380739
#### Screenshots (if appropriate)
N/A
